sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/screens/home/HomeViewModel.kt

```diff
@@ -12,18 +12,24 @@ import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
-    private val useCase: UseCase,
+    private val getModelsUseCase: GetModelsUseCase,
+    private val isFirstTimeLaunchPreferencesUseCase: IsFirstTimeLaunchPreferencesUseCase,
+    private val updateFirstTimeLaunchPreferencesUseCase: UpdateFirstTimeLaunchPreferencesUseCase,
     dispatchers: DispatchersProvider
 ) : BaseViewModel(dispatchers) {
 
     private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
     val uiModels: StateFlow<List<UiModel>>
         get() = _uiModels
 
+    private val _isFirstTimeLaunch = MutableStateFlow(false)
+    val isFirstTimeLaunch: StateFlow<Boolean>
```

Comment:
@lydiasama We can apply using SharedFlow to avoid duplicated Toast showing on the Home screen.

Reply:
@luongvo I wonder if it's the issue of emitting several times. Since I debugged as: 
> The toast message from my POW might show twice. But after debugging, I double-checked it emits the event just once.

I found that it emits just once which is correct. 🤔 And also in sample compose, we use `StateFlow` as well. What do you think?

Reply:
@lydiasama Using `StateFlow` will re-trigger to bind the current value when `bindViewModel()` is recalled. That's why we used `SharedFlow` to display the `error message`, but `StateFlow` for `Loading` because re-showing the error message does not make sense, however, re-updating loading progress visibility is good. 

We might don't have the same issue in Compose template & sample projects because we use different ways to bind & stream data from VM to Composable.

Reply:
To have a better explanation of this issue, say your HomeViewModel has `error as StateFlow` with pre-emitted Throwable value, going back to HomeFragment from another fragment will re-trigger `bindViewModel()` and re-showing the error message 💥